### PR TITLE
REPO-3680: Renditions: Rendition Service can call new Transform Service

### PR DESCRIPTION
**Pull request by cristianturlica**
- added send after commit support

### CODE REVIEW

File: src/main/java/org/alfresco/repo/rawevents/EventProducer.java [hunk 1]
```diff
@@ -51,8 +59,12 @@
  */
 public class EventProducer
 {
+    private static final String POST_TRANSACTION_PENDING_REQUESTS = "postTransactionPendingEventRequests";
     protected static final String ERROR_SENDING = "Could not send event";
 
+    private static Log logger = LogFactory.getLog(EventProducer.class);
+
+    private TransactionListener transactionListener = new TransactionListener();
```
Comment (killerboot) [inline src/main/java/org/alfresco/repo/rawevents/EventProducer.java]: It is the same bug (and lot's of code duplication), that was introduced in this [branch](https://github.com/Alfresco/alfresco-repository/tree/feature/repo-3676-another-approach) in RendtionService2Impl.

The same tx listener will be bound to all tx. That may result in concurrency issues.
I corrected it in a refactoring, see 68d66b1905dbbcaf38bbba5957cd1eb39139eeae. This needs to go through testing. But the new RenditionRequestScheduler can be generic enough to execute any post transaction code.

File: src/test/java/org/alfresco/repo/rawevents/EventProducerTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Provides a base set of tests for {@link EventProducer}
+ *
+ * @author Cristian Turlica
+ */
+@Category(OwnJVMTestsCategory.class)
+public class EventProducerTest
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/rawevents/EventProducerTest.java]: Is this included in any test suits?

Comment (Epurashu) [reply src/test/java/org/alfresco/repo/rawevents/EventProducerTest.java]: It appears to not be part of any test suite, please add it 👍 .

Comment (cristianturlica) [reply src/test/java/org/alfresco/repo/rawevents/EventProducerTest.java]: Changed, thanks.

**Pull request merged**